src/AspNet.Security.OAuth.Bitrix24/Bitrix24AuthenticationOptions.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Globalization;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.Bitrix24.Bitrix24AuthenticationConstants;
+
+namespace AspNet.Security.OAuth.Bitrix24
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="Bitrix24AuthenticationHandler"/>.
+    /// </summary>
+    public class Bitrix24AuthenticationOptions : OAuthOptions
+    {
+        private string? tenantHostName;
+
+        public string? TenantHostName
+        {
+            get
+            {
+                return tenantHostName;
+            }
+
+            set
+            {
+                tenantHostName = value;
+                AuthorizationEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.AuthorizationEndpoint, value);
+                TokenEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.TokenEndpoint, value);
+                UserInformationEndpoint = string.Format(CultureInfo.InvariantCulture, Bitrix24AuthenticationDefaults.UserInformationEndpoint, value);
+            }
+        }
```

Comment:
As noted earlier, refactor this to use the pattern the Okta provider (and others) use for varying host names.